### PR TITLE
8248186: Move CDS C++ vtable code to cppVtables.cpp

### PR DESCRIPTION
**Pull request by iklam**
I moved the code that supports C++ vtables in the CDS archive from metaspaceShared.cpp to a new file, cppVtables.cpp. To keep the refactoring straightforward, the code is moved verbatim, except for:

- Methods are renamed from `MetaspaceShared::xx` to `CppVtables::xx`
- Access to `_mc_region` is changed to the inline function `mc_region()`

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8248186](https://bugs.openjdk.java.net/browse/JDK-8248186): Move CDS C++ vtable code to cppVtables.cpp

### Reviewers
 * [Coleen Phillimore](https://openjdk.java.net/census#coleenp) (@coleenp - **Reviewer**) ⚠️ Review applies to b1db967d22ec74d1c62504c936b5f59c48a45bc3

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/136/head:pull/136`
`$ git checkout pull/136`

### CODE REVIEW

**Comment by iklam** [comment]
label add hotspot-runtime

File: src/hotspot/share/memory/cppVtables.cpp [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+
+#include "precompiled.hpp"
+#include "logging/log.hpp"
+#include "memory/archiveUtils.hpp"
+#include "memory/cppVtables.hpp"
+#include "memory/metaspaceShared.hpp"
+#include "oops/instanceClassLoaderKlass.hpp"
+#include "oops/instanceMirrorKlass.hpp"
+#include "oops/instanceRefKlass.hpp"
+#include "oops/methodData.hpp"
```
Comment (coleenp) [inline src/hotspot/share/memory/cppVtables.cpp:33]: Did you check that you needed all of these #include files?

**Review by coleenp — APPROVED** [review]

File: src/hotspot/share/memory/cppVtables.cpp (hunk 1 above)
Comment (iklam) [reply src/hotspot/share/memory/cppVtables.cpp:33]: Yes, I started with no headers and kept adding until gcc stopped complaining. I was using a non-PCH build.

**Pull request closed without merging**